airflow/settings.py

```diff
@@ -251,16 +251,24 @@ def prepare_classpath():
         sys.path.append(PLUGINS_FOLDER)
 
 
-try:
-    from airflow_local_settings import *  # noqa F403 F401
-    log.info("Loaded airflow_local_settings.")
-except Exception:
-    pass
+def import_local_settings():
+    import airflow_local_settings
+
+    for k, v in airflow_local_settings.__dict__.items():
+        if not k.startswith("__") and not k.endswith("__"):
+            globals()[k] = v
```

Comment:
Do we need to change this? `from airflow_local_settings import *` respects `__all__` if you wish to control what is imported. http://xion.io/post/code/python-all-wild-imports.html

Reply:
Unfortunately yes, because wild imports are not allow at the function level: `SyntaxError: import * only allowed at module level`.

The logic on line 258 could be improved though, but I'm not sure what would work best.

Reply:
Ohhh. Yeah :(

Reply:
We should possibly respect `__all__` if it's set then.

Reply:
One thing I am unsure about, should we exclude just magic attributes or anything that starts with `_`, by convention these are supposedly non-public?

Reply:
Just exclude `__` - `_foo`  would still be imported by a `from mod import *` and that's what we should emulate here.

Reply:
Added `__all__` support and associated test.